### ISSUE TITLE
Improve microsite

### ISSUE THREAD

**Issue by BalmungSan** [neotypes/neotypes#252]
It would be good to improve the microsite so it provides a better experience for our users.
Right now I can think about the following things:

1. Use the new _(and recommended)_ `light` theme, instead of the old `pattern` one. I believe this force us to remove our current custom styles, so we would need to see if we like the new theme or see how to customize it.
2. Remove the current `index.md` and rather use the `README.md`, that way we do not need to maintain both. Also, it would be good to preserve the **Compatibility matrix** that currently is not in the `README`.
3. Make the `README` / `index.md` have the current latest version in the **Setup** section instead of saying _"version"_.
4. Adding new sections? It would be good if users could point out what they find missing in the current docs.

**Comment by i10416** [comment]
I think Scala 3's scaladoc([api docs + blog](https://docs.scala-lang.org/scala3/guides/scaladoc/blog.html)) and [snippet checking](https://docs.scala-lang.org/scala3/guides/scaladoc/snippet-compiler.html) are the nice alternative for current approach.
sbt-typelevel seems nice too especially when you want to manage complex build matrix(scala 2 and 3, scala JVM, JS and Native).

**Comment by BalmungSan** [comment]
@i10416 yeah, at some point, we really should just use **sbt-typelevel** and migrate everything to what they use.
